features/dd-sdk-android-session-replay-compose/src/main/kotlin/com/datadog/android/sessionreplay/compose/internal/mappers/semantics/CheckboxSemanticsNodeMapper.kt

```diff
@@ -0,0 +1,205 @@
+        asyncJobStatusCallback: AsyncJobStatusCallback,
+        semanticsNode: SemanticsNode,
+        globalBounds: GlobalBounds
+    ): List<MobileSegment.Wireframe> {
+        val rawFillColor = semanticsUtils.resolveCheckboxFillColor(semanticsNode)
+        val rawCheckmarkColor = semanticsUtils.resolveCheckmarkColor(semanticsNode)
+
+        val fillColorRgba = rawFillColor?.let {
+            convertColor(it)
+        } ?: DEFAULT_COLOR_WHITE
+        val checkmarkColorRgba = rawCheckmarkColor?.let {
+            convertColor(it)
+        } ?: getFallbackCheckmarkColor(DEFAULT_COLOR_WHITE)
+
+        val parsedFillColor = pathUtils.parseColorSafe(fillColorRgba)
+        val parsedCheckmarkColor = pathUtils.parseColorSafe(checkmarkColorRgba)
+
+        if (parsedFillColor != null && parsedCheckmarkColor != null) {
+            val checkMarkBitmap = semanticsUtils
+                .resolveCheckPath(semanticsNode)?.let { checkPath ->
+                    pathUtils.convertPathToBitmap(
+                        checkPath = checkPath,
+                        fillColor = parsedFillColor,
+                        checkmarkColor = parsedCheckmarkColor
+                    )
+                }
+
+            if (checkMarkBitmap != null) {
+                parentContext.imageWireframeHelper.createImageWireframeByBitmap(
+                    id = resolveId(semanticsNode, 0),
+                    globalBounds = globalBounds,
+                    bitmap = checkMarkBitmap,
+                    density = parentContext.density,
+                    isContextualImage = false,
+                    imagePrivacy = ImagePrivacy.MASK_NONE,
+                    asyncJobStatusCallback = asyncJobStatusCallback,
+                    clipping = null,
+                    shapeStyle = MobileSegment.ShapeStyle(
+                        backgroundColor = fillColorRgba,
+                        opacity = 1f,
+                        cornerRadius = CHECKBOX_CORNER_RADIUS
+                    ),
+                    border = MobileSegment.ShapeBorder(
+                        color = fillColorRgba,
+                        width = BOX_BORDER_WIDTH_DP
+                    )
+                )?.let { imageWireframe ->
+                    return listOf(imageWireframe)
+                }
+            }
+        }
+
+        // if we failed to create a wireframe from the path
+        return createManualCheckedWireframe(semanticsNode, globalBounds, fillColorRgba)
+    }
+
+    private fun createManualCheckedWireframe(
+        semanticsNode: SemanticsNode,
+        globalBounds: GlobalBounds,
+        backgroundColor: String
+    ): List<MobileSegment.Wireframe> {
+        val strokeColor = getFallbackCheckmarkColor(backgroundColor)
+
+        val background: MobileSegment.Wireframe = createUncheckedWireframes(
+            semanticsNode = semanticsNode,
+            globalBounds = globalBounds,
+            backgroundColor = backgroundColor
+        )[0]
```

Comment:
It's guaranteed that it won't be empty - createUncheckedWireframes always returns a result. However I'll modify the method to not return a list and wrap with the list externally as it may make it clearer